indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -384,7 +383,12 @@ public static DataSegment serializeOutIndex(
       throws IOException
   {
     final FileSystem outputFS = FileSystem.get(segmentBasePath.toUri(), configuration);
-    final Path tmpPath = new Path(segmentBasePath, String.format("index.zip.%d", taskAttemptID.getId()));
+
+    final Path tmpPath = outputFS.getScheme().equals("hdfs") || outputFS.getScheme().equals("viewfs") ? new Path(
+        segmentBasePath,
+        String.format("%d_index.zip.%d", segmentTemplate.getShardSpec().getPartitionNum(), taskAttemptID.getId())
+    ) : new Path(segmentBasePath, String.format("index.zip.%d", taskAttemptID.getId()));
```

Comment:
This conditional for tmpPath isn't necessary. You could use the "0_index.zip.taskAttemptId" format for all schemes and it would work fine, and be simpler.